### PR TITLE
Write version check warning message to stderr

### PR DESCRIPTION
**Pull request by weikanglim**
Quick fix for #683

### CODE REVIEW

File: cli/azd/main.go [hunk 1]
```diff
@@ -63,14 +63,14 @@ func main() {
 			// This is a dev build (i.e. built using `go install without setting a version`) - don't print a warning in this case
 			log.Printf("eliding update message for dev build")
 		} else if latestVersion.GT(curVersion) {
-			fmt.Printf(output.WithWarningFormat("warning: your version of azd is out of date, you have %s and the latest version is %s\n"), curVersion.String(), latestVersion.String())
-			fmt.Println()
-			fmt.Println(output.WithWarningFormat(`To update to the latest version, run:`))
+			fmt.Fprintf(os.Stderr, "warning: your version of azd is out of date, you have %s and the latest version is %s\n", curVersion.String(), latestVersion.String())
```
Comment (ellismg) [inline cli/azd/main.go:66]:
```suggestion
			fmt.Fprintf(os.Stderr, output.WithWarningFormat("warning: your version of azd is out of date, you have %s and the latest version is %s\n", curVersion.String(), latestVersion.String()))
```

**Review by ellismg — APPROVED** [review]

**Comment by karolz-ms** [comment]
The only potential problem with this is that older versions of PowerShell would consider a command execution to have failed if anything showed up on `stderr`, regardless of exit code. But maybe newer PowerShell Core does not suffer from this--can you double-check, @weikanglim ?

File: cli/azd/main.go (hunk 1 above)
Comment (weikanglim) [reply cli/azd/main.go:66]: Was wondering why I was missing a parenthesis. Good catch..

**Comment by weikanglim** [comment]
> The only potential problem with this is that older versions of PowerShell would consider a command execution to have failed if anything showed up on `stderr`, regardless of exit code. But maybe newer PowerShell Core does not suffer from this--can you double-check, @weikanglim ?

I'll double check on the behavior.

**Comment by weikanglim** [comment]
@karolz-ms Can confirm the behavior documented in https://stackoverflow.com/questions/66726049/how-can-i-redirect-stdout-and-stderr-without-polluting-powershell-error-output. On Windows PowerShell 5.X or before PowerShell 7.2, the act of redirecting the error stream in PowerShell causes a terminating error to be detected. This no longer happens in the PowerShell version 7.2 or greater.

Test script `test.ps1`:
```pwsh
param(
    [switch]$RedirectError
)
$ErrorActionPreference = 'Stop'

if ($RedirectError) {
    .\azd.exe version 2>&1
} else {
    .\azd.exe version
}

Write-Host 'SUCCESS'
```

On 7.2.6:
```pwsh
PS D:\repos\azd-sources\demo-todo-python> $PSVersionTable.PSVersion.ToString()
7.2.6
PS D:\repos\azd-sources\demo-todo-python> .\test.ps1
azd version 0.1.0-dev.0 (commit 0000000000000000000000000000000000000000)
warning: your version of azd is out of date, you have 0.1.0-dev.0 and the latest version is 0.2.0-beta.1

To update to the latest version, run:
powershell -ex AllSigned -c "Invoke-RestMethod 'https://aka.ms/install-azd.ps1' | Invoke-Expression"
SUCCESS
PS D:\repos\azd-sources\demo-todo-python> .\test.ps1 -RedirectError
azd version 0.1.0-dev.0 (commit 0000000000000000000000000000000000000000)
warning: your version of azd is out of date, you have 0.1.0-dev.0 and the latest version is 0.2.0-beta.1

To update to the latest version, run:
powershell -ex AllSigned -c "Invoke-RestMethod 'https://aka.ms/install-azd.ps1' | Invoke-Expression"
SUCCESS
```

On 5.X:
```pwsh
PS D:\repos\azd-sources\demo-todo-python>  $PSVersionTable.PSVersion.ToString()
5.1.19041.1682
PS D:\repos\azd-sources\demo-todo-python> .\test.ps1
azd version 0.1.0-dev.0 (commit 0000000000000000000000000000000000000000)
warning: your version of azd is out of date, you have 0.1.0-dev.0 and the latest version is 0.2.0-beta.1

To update to the latest version, run:
powershell -ex AllSigned -c "Invoke-RestMethod 'https://aka.ms/install-azd.ps1' | Invoke-Expression"
SUCCESS
PS D:\repos\azd-sources\demo-todo-python> .\test.ps1 -RedirectError
azd version 0.1.0-dev.0 (commit 0000000000000000000000000000000000000000)
.\azd.exe : warning: your version of azd is out of date, you have 0.1.0-dev.0 and the latest version is 0.2.0-beta.1
At D:\repos\azd-sources\demo-todo-python\test.ps1:7 char:5
+     .\azd.exe version 2>&1
+     ~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (warning: y....2.0-beta.1:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError

```

**Comment by karolz-ms** [comment]
@weikanglim glad to her, thanks for confirming!!

**Pull request merged**